src/source-dpdk.h

```diff
@@ -0,0 +1,91 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Lukas Sismis <lukas.sismis@gmail.com>
+ */
+
+#ifndef __SOURCE_DPDK_H__
+#define __SOURCE_DPDK_H__
+
+/* Maximum DPDK EAL parameters count. */
+#define EAL_ARGS 32
+
+#include "queue.h"
+
+#ifdef HAVE_DPDK
+#include <rte_ethdev.h>
+#endif
+
+typedef enum { DPDK_COPY_MODE_NONE, DPDK_COPY_MODE_TAP, DPDK_COPY_MODE_IPS } DpdkCopyModeEnum;
+
+/* DPDK Flags */
+// General flags
+#define DPDK_PROMISC   (1 << 0) /**< Promiscuous mode */
+#define DPDK_MULTICAST (1 << 1) /**< Enable multicast packets */
+// Offloads
+#define DPDK_RX_CHECKSUM_OFFLOAD (1 << 4) /**< Enable chsum offload */
+
+typedef struct DPDKIfaceConfig_ {
+#ifdef HAVE_DPDK
+    char iface[RTE_ETH_NAME_MAX_LEN];
+    uint16_t port_id;
+    uint16_t socket_id;
+    /* number of threads - zero means all available */
+    int threads;
+    /* IPS mode */
+    DpdkCopyModeEnum copy_mode;
```

Comment:
in IPS mode you are not copying packets from the Ingress interface, It is in IDS mode one needs to mirror the packet (if required). But normally IDS is deployed on mirrored traffic, if are you planning to mirror the packet there is no code found to mirror via SW or using HW. please rework.

Reply:
I think there is a slight misunderstanding here. During my development, I tried to follow AF_PACKET capture runmode to keep consistent with the architecture/naming conventions. This field does not in fact represent packet mirroring. [The same way it is used in AF_PACKET](https://github.com/OISF/suricata/blob/b8499de498f0e31bb31097ac8b959e3877573de0/src/source-af-packet.h#L68). This field holds information whether Suricata is in None (IDS) / TAP(IPS) / True IPS mode. 
And if Suricata is in either TAP or IPS mode, the packets are transmitted in packet release function. ([like here](https://github.com/OISF/suricata/blob/b8499de498f0e31bb31097ac8b959e3877573de0/src/source-af-packet.c#L828)). If Suricata is in IDS mode, the packet is only released.
So to wrap up, there is no packet mirroring and copy_mode only represents the mode Suricata is in. DPDK runmode does not copy packets anywhere. The name was only derived from AF_PACKET implementation to keep implementations consistent.